libdnf/dnf-repo.c

```diff
@@ -1215,7 +1339,13 @@ dnf_repo_check_internal(DnfRepo *repo,
         if (!dnf_repo_set_timestamp_modified(repo, error))
             return FALSE;
         age_of_data =(g_get_real_time() - priv->timestamp_modified) / G_USEC_PER_SEC;
-        if (age_of_data > permissible_cache_age) {
+        metadata_expire = dnf_repo_get_metadata_expire(repo);
+        if(metadata_expire != 0)
+            valid_time_allowed = metadata_expire <= permissible_cache_age ? metadata_expire  : permissible_cache_age;
```

Comment:
Hmm.  I'm not sure we need `permissible_cache_age` anymore; we can just have it be a default value for `metadata_expire` right?